pkg/alertmanager/api.go

```diff
@@ -151,7 +156,8 @@ func validateUserConfig(cfg alerts.AlertConfigDesc) error {
 	for _, tmpl := range cfg.Templates {
 		_, err := createTemplateFile(tmpDir, cfg.User, tmpl.Filename, tmpl.Body)
 		if err != nil {
-			return err
+			level.Error(logger).Log("msg", "unable to create template file", "err", err, "user", cfg.User)
+			return fmt.Errorf("unable to create template file %s", tmpl.Filename)
```

Comment:
I had to re-word this error has it was providing information about the FS - I don't think it could have been an attack vector but it felt like it had too much information in it.

Reply:
I think it can, and if tmpl.Filename contains eg. `hello/../../../../../etc/passwd` it could possibly rewrite `/etc/password` even now (if permissions allow). I don't see any checks for path traversal.

Reply:
Done! WDYT?

Reply:
Looks good, thanks. I'm wondering if we should put that check into `createTemplateFile` instead, since it's being called from multiple places?

Reply:
This call is only internal and getting a useful error message would require some effort if we put inside the function given we're not taking the error message into account - not sure it's worth the effort at this point.

Reply:
Done as part of https://github.com/cortexproject/cortex/pull/3185/commits/f731b9ab85e5ba3a49da4fdda4811eeb208cfba2